arch/riscv/Kconfig

```diff
@@ -84,6 +84,14 @@ config RISCV_SOC_HAS_ISR_STACKING
 	    saved on the stack by the hardware, and the registers saved by the
 	    software macros. The structure must be called '__esf'.
 
+config RISCV_SOC_HAS_CUSTOM_IRQ_HANDLING
+	bool
+	default y if NUCLEI_ECLIC && !RISCV_VECTORED_MODE
```

Comment:
Hi! I'm not entirely convinced that adding a default y-selection of this symbol here, i.e. essentially selecting arch-related symbols by a driver, is a good idea.

What about y-selecting `RISCV_SOC_HAS_CUSTOM_IRQ_HANDLING` in https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/riscv/gd_gd32/gd32vf103/Kconfig.defconfig.gd32vf103 instead?

Reply:
I'm not sure, whether that's the best place to put it, because it is not specific only to the gd32v. The option should be enabled for all ECLICs in non-vectored mode.

Maybe I could y-select it from `config NUCLEI_ECLIC` in drivers/interrupt_controller/Kconfig.clic, so 

```
config NUCLEI_ECLIC
    <...>
    select RISCV_SOC_HAS_CUSTOM_IRQ_HANDLING if !RISCV_VECTORED_MODE
```

Reply:
@fkokosinski I've implemented it now according to my proposal. Is this acceptable?

Reply:
Thanks, looks good